code/game/objects/items/plushes.dm

```diff
@@ -599,6 +599,28 @@
 	icon_state = "goat"
 	desc = "Despite its cuddly appearance and plush nature, it will beat you up all the same. Goats never change."
 	squeak_override = list('sound/weapons/punch1.ogg'=1)
+	var/going_hard = FALSE
+
+/obj/item/toy/plush/goatplushie/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/clothing/mask/cigarette/rollie))
+		var/obj/item/clothing/mask/cigarette/rollie/fat_dart = I
+		if(!going_hard)
+			if(fat_dart.lit)
+				to_chat(user, span_notice("You put [fat_dart] into [src]'s mouth."))
+				qdel(fat_dart)
+				add_overlay("goat_dart")
```

Comment:
this will go away when overlays are refreshed, instead put this on update_overlays and update_overlays after setting `going_hard`

Reply:
I tried doing that before, it didn't add the overlay at all when I did.
I'd previously had `. += "goat_dart"` in `update_overlays()`

Reply:
I put it back how I had it before, still nothing.

Reply:
Got it!